tests/types/test_resolver_types.py

```diff
@@ -0,0 +1,104 @@
+from enum import Enum
+from typing import List, Optional, TypeVar, Union
+
+import strawberry
+from strawberry.types.fields.resolver import StrawberryResolver
+
+
+def test_enum():
+    @strawberry.enum
+    class Language(Enum):
+        ENGLISH = "english"
+        ITALIAN = "italian"
+        JAPANESE = "japanese"
+
+    def get_spoken_language() -> Language:
+        return Language.ENGLISH
+
+    resolver = StrawberryResolver(get_spoken_language)
+    # TODO: Remove reference to ._enum_definition with StrawberryEnum
+    assert resolver.type is Language._enum_definition
+
+
+def test_forward_references():
+    global FutureUmpire
+
+    def get_sportsball_official() -> "FutureUmpire":
+        return FutureUmpire("ref")
+
+    @strawberry.type
+    class FutureUmpire:
+        name: str
+
+    resolver = StrawberryResolver(get_sportsball_official)
+    assert resolver.type is FutureUmpire
+
+    del FutureUmpire
+
+
+def test_lazy_type():
+    # TODO: Implement this test
+    ...
+
+
+def test_list():
+    def get_collection_types() -> List[str]:
+        return ["list", "tuple", "dict", "set"]
+
+    resolver = StrawberryResolver(get_collection_types)
+    assert resolver.type == List[str]
+
+
+def test_literal():
+    def version() -> float:
+        return 1.0
+
+    resolver = StrawberryResolver(version)
+    assert resolver.type is float
+
+
+def test_object():
+    @strawberry.type
+    class Polygon:
+        edges: int
+        faces: int
+
+    def get_2d_object() -> Polygon:
+        return Polygon(12, 6)
+
+    resolver = StrawberryResolver(get_2d_object)
+    assert resolver.type is Polygon
+
+
+def test_optional():
+    def stock_market_tool() -> Optional[str]:
+        ...
+
+    resolver = StrawberryResolver(stock_market_tool)
+    assert resolver.type == Optional[str]
+
+
+def test_type_var():
+    T = TypeVar("T")
+
+    def caffeinated_drink() -> T:
+        ...
+
+    resolver = StrawberryResolver(caffeinated_drink)
+    assert resolver.type == T
+
+
+def test_union():
+    @strawberry.type
+    class Venn:
+        foo: int
+
+    @strawberry.type
+    class Diagram:
+        bar: float
+
+    def get_overlap() -> Union[Venn, Diagram]:
+        ...
+
+    resolver = StrawberryResolver(get_overlap)
+    assert resolver.type == Union[Venn, Diagram]
```

Comment:
do we test calling resolvers?

Reply:
I have added no tests calling resolvers. Is there something specific to this change that would warrant additions?

Reply:
no, just wanted if you wanted to test calling a strawberry resolver directly :)

Reply:
you mean
```python
resolver = StrawberryResolver(...)
resolver()
```
?

Reply:
yup 😊

Reply:
I'll open an issue for that, but the logic is very simple.
https://github.com/strawberry-graphql/strawberry/blob/5b601be026a329ed5e04f48b2291b311efa79520/strawberry/types/fields/resolver.py#L21-L22

Reply:
https://github.com/strawberry-graphql/strawberry/issues/1067